CoffeeShop.js

```diff
@@ -0,0 +1,106 @@
+class File {
+	static Exists(filePath) {
+		if (require("fs").existsSync(filePath))
+			return true;
+
+		return false;
+	}
+
+	static ReadAllLines(filePath) {
+		return require("fs").readFileSync(filePath).toString().split('\n')
+	}
+}
+
+function ParseCSV(filePath) {
+	var rawData = File.ReadAllLines(filePath);
+	var arrData = Array.from(Array(rawData.length), () => new Array(3));
+
+	try {
+		for (var i = 0; i < rawData.length; i++) {
+			if ((rawData[i].split(',').length - 1) != 2)
+				throw new Error(`Corrupted CSV template (check for the commas) -> row [${i + 1}]`);
+
+			// Retrieves the coffee shop name
+			arrData[i][0] = rawData[i].slice(0, rawData[i].indexOf(','));
+
+			// Retrieves the X user coordinate
+			arrData[i][1] = rawData[i].slice(rawData[i].indexOf(',') + 1, rawData[i].lastIndexOf(','));
+
+			if (isNaN(arrData[i][1]))
+				throw new Error(`Invalid [X] coordinate -> row [${i + 1}]`);
+			// Retrieves the Y user coordinate
+			arrData[i][2] = rawData[i].slice(rawData[i].lastIndexOf(',') + 1);
+
+			if (isNaN(arrData[i][2]))
+				throw new Error(`Invalid [Y] coordinate -> row [${i + 1}]`);
+		}
+	} catch(e) {
+		throw new Error(`Can't parse the selected CSV file!\n   > : {${e}}`);
+	}
+
+	return arrData;
+}
+
+function CalculateDistance(x, y, x1, y1) {
+  const p = 0.017453292519943295; // Math.PI / 180
+  const c = Math.cos;
+  const a = 0.5 - c((x1 - x) * p) / 2 + 
+          c(x * p) * c(x1 * p) * 
+          (1 - c((y1 - y) * p)) / 2;
+
+  return 12742 * Math.asin(Math.sqrt(a)); // 2 * R; R = 6371 km
+}
+
+
+function Main() {
+	var user_xcoo         = 0.00; // User X coordinates
+	var user_ycoo         = 0.00; // User Y coordinates
+	var shopCsv_filePath  = "";   // Shops CSV local file path
+	var shopCsv_fileData;         // Shops CSV file data
+
+	// # ARGUMENT PARSING # //
+	process.argv.forEach(function(val, index, array) {
+		switch(index) {
+			case 2:
+				user_xcoo = parseFloat(val);
```

Comment:
https://www.smashingmagazine.com/2012/10/why-coding-style-matters/

Every language and project has a coding convention. Example: https://github.com/airbnb/javascript
This can be enforced with linting tools (eslint)

Reply:
Thank you for sharing this information with me, I'll read everything in order to improve my skills